ui/src/app/machines/views/AddMachine/AddMachineForm/AddMachineForm.js

```diff
@@ -0,0 +1,224 @@
+import { Loader } from "@canonical/react-components";
+import React, { useEffect, useMemo, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import * as Yup from "yup";
+
+import {
+  domain as domainActions,
+  general as generalActions,
+  machine as machineActions,
+  resourcepool as resourcePoolActions,
+  zone as zoneActions
+} from "app/base/actions";
+import {
+  domain as domainSelectors,
+  general as generalSelectors,
+  machine as machineSelectors,
+  resourcepool as resourcePoolSelectors,
+  zone as zoneSelectors
+} from "app/base/selectors";
+import { useAddMessage, useWindowTitle } from "app/base/hooks";
+import AddMachineFormFields from "../AddMachineFormFields";
+import FormCard from "app/base/components/FormCard";
+import FormikForm from "app/base/components/FormikForm";
+import FormCardButtons from "app/base/components/FormCardButtons";
+
+const trimPowerParameters = (powerType, powerParameters) => {
```

Comment:
Might be useful to leave a comment here about why we need to do this.

Reply:
Good idea. Added it in now.